src/runtime/pay-flow.js

```diff
@@ -126,7 +126,15 @@ export class PayStartFlow {
     }
 
     // Start/cancel events.
-    this.deps_.callbacks().triggerFlowStarted(SubscriptionFlows.SUBSCRIBE, req);
+    if (this.productType_ == ProductType.SUBSCRIPTION) {
```

Comment:
Reduce all this duplication in code.  E.g. on line 127, add 

`const flow = this.productType_ == ProductType.UI_CONTRIBUTION ? SubscriptionFlows.CONTRIBUTE : SubscriptionFlows.SUBSCRIBE;
`

Then simply call `this.deps_().callbacks().triggerFlowStarted(flow, req);`